profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PprofAllocationEventExporter.java

```diff
@@ -0,0 +1,291 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.profiler.allocation.exporter;
+
+import static com.google.perftools.profiles.ProfileProto.*;
+import static com.splunk.opentelemetry.profiler.LogDataCreator.PROFILING_SOURCE;
+import static com.splunk.opentelemetry.profiler.LogExporterBuilder.INSTRUMENTATION_LIBRARY_INFO;
+import static com.splunk.opentelemetry.profiler.ProfilingSemanticAttributes.SOURCE_TYPE;
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+
+import com.splunk.opentelemetry.profiler.Configuration.AllocationDataFormat;
+import com.splunk.opentelemetry.profiler.allocation.sampler.AllocationEventSampler;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.sdk.logs.LogProcessor;
+import io.opentelemetry.sdk.logs.data.LogDataBuilder;
+import io.opentelemetry.sdk.resources.Resource;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.zip.GZIPOutputStream;
+import jdk.jfr.consumer.RecordedEvent;
+import jdk.jfr.consumer.RecordedFrame;
+import jdk.jfr.consumer.RecordedMethod;
+import jdk.jfr.consumer.RecordedStackTrace;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PprofAllocationEventExporter implements AllocationEventExporter {
+  private static final Logger logger = LoggerFactory.getLogger(PprofAllocationEventExporter.class);
+
+  private final LogProcessor logProcessor;
+  private final Resource resource;
+  private final AllocationDataFormat allocationDataFormat;
+  private ProfileData profileData = new ProfileData();
+
+  private PprofAllocationEventExporter(Builder builder) {
+    this.logProcessor = builder.logProcessor;
+    this.resource = builder.resource;
+    this.allocationDataFormat = builder.allocationDataFormat;
+  }
+
+  @Override
+  public void export(RecordedEvent event, AllocationEventSampler sampler, SpanContext spanContext) {
+    RecordedStackTrace stackTrace = event.getStackTrace();
+    if (stackTrace == null) {
+      return;
+    }
+
+    Profile.Builder profile = profileData.profile;
+    LocationTable locationTable = profileData.locationTable;
+    StringTable stringTable = profileData.stringTable;
+
+    // XXX jdk mission control "TLAB allocations" view seems to use tlabSize as allocation size
```

Comment:
Agent use `allocationSize`. I added this comment just as a note that there is also `tlabSize` in case you wan't to estimate allocation rate where `tlabSize` would give a much better result. Though sampling these events would still make it useless.

Reply:
Yeah. We get a very good number for allocation rate from the `jvm.experimental.memory.allocated` metric, and these tlab events represent like 2% of all allocations in my experiments. So I think we can skip this.